helm-chart/templates/nginx-mesh-api.yaml

```diff
@@ -181,14 +181,85 @@ metadata:
     app.kubernetes.io/part-of: nginx-service-mesh
     spiffe.io/webhook: "true"
 webhooks:
-- name: nginx-mesh-api.sidecar.injector
+- name: nginx-mesh-api-ns.sidecar.injector
   namespaceSelector:
     matchExpressions:
     - key: kubernetes.io/metadata.name
       operator: NotIn
       values:
       - kube-system
       - {{ .Release.Namespace }}
+    matchLabels:
+      injector.nsm.nginx.com/auto-inject: enabled
+  clientConfig:
+    service:
+      name: nginx-mesh-webhook
+      namespace: {{ .Release.Namespace }}
+      path: /inject
+  sideEffects: None
+  admissionReviewVersions: ["v1"]
+  rules:
+  - apiGroups: [""]
+    apiVersions: ["v1"]
+    operations: ["CREATE"]
+    resources: ["pods"]
+- name: nginx-mesh-api-obj.sidecar.injector
+  namespaceSelector:
+    matchExpressions:
+    - key: kubernetes.io/metadata.name
+      operator: NotIn
+      values:
+      - kube-system
+      - {{ .Release.Namespace }}
+  objectSelector:
+      matchLabels:
+       injector.nsm.nginx.com/auto-inject: enabled
+  clientConfig:
+    service:
+      name: nginx-mesh-webhook
+      namespace: {{ .Release.Namespace }}
+      path: /inject
+  sideEffects: None
+  admissionReviewVersions: ["v1"]
+  rules:
+  - apiGroups: [""]
+    apiVersions: ["v1"]
+    operations: ["CREATE"]
+    resources: ["pods"]
+- name: nginx-mesh-api-ingress.sidecar.injector
+  namespaceSelector:
+    matchExpressions:
+    - key: kubernetes.io/metadata.name
+      operator: NotIn
+      values:
+      - kube-system
+      - {{ .Release.Namespace }}
+  objectSelector:
+    matchLabels:
+      nsm.nginx.com/enable-ingress: "true"
+  clientConfig:
+    service:
+      name: nginx-mesh-webhook
+      namespace: {{ .Release.Namespace }}
+      path: /inject
+  sideEffects: None
+  admissionReviewVersions: ["v1"]
+  rules:
+  - apiGroups: [""]
+    apiVersions: ["v1"]
+    operations: ["CREATE"]
+    resources: ["pods"]
+- name: nginx-mesh-api-egress.sidecar.injector
+  namespaceSelector:
```

Comment:
I would like to see comments similar to Saylor's linked example added. That way the intention of each set of selectors is clearly expressed.

Something similar to this:
`{{- /* Case 1: namespace selector matches, and object doesn't disable */}}`
`{{- /* Note: if both revision and legacy selector, we give precedence to the legacy one */}}`